### PR TITLE
core: docs improvements

### PR DESCRIPTION
**Pull request by coryrylan**
## PR Checklist

Please check if your PR fulfills the following requirements:

* [ ] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [x] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the new behavior?

- Adds minimal docs to show import path and element usage for core components.
- Tags all core components as beta for 3.0
- Update attrs to default as props as reflect back to get better static typing support across frameworks

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

### CODE REVIEW

**Comment by coryrylan** [comment]
Added a small fix for base button to get in before end of day.

File: src/clr-core/badge/badge.element.ts [hunk 1]
```diff
@@ -40,6 +46,14 @@ import { styles } from './badge.element.css';
  */
 // @dynamic
 export class CwcBadge extends LitElement {
+  /** Sets the color of the badge from a predefined list of choices */
+  @property({ type: String, reflect: true })
+  color: 'gray' | 'purple' | 'blue' | 'orange' | 'light-blue';
+
+  /** Sets the color of the badge from a predefined list of statuses */
+  @property({ type: String, reflect: true })
+  status: 'info' | 'success' | 'warning' | 'danger';
```
Comment (mathisscott) [inline src/clr-core/badge/badge.element.ts]: Not going to block on this. But it seems like we could turn `'info' | 'success' | 'warning' | 'danger'` into its own type. We use it everywhere...

File: src/clr-core/tag/tag.element.ts [hunk 2]
```diff
@@ -49,6 +56,14 @@ import { styles } from './tag.element.css';
  */
 // @dynamic
 export class CwcTag extends CwcBaseButton {
+  /** Sets the color of the tag (and badge if present) from a predefined list of statuses */
+  @property({ type: String, reflect: true })
+  status: 'info' | 'success' | 'warning' | 'dander';
```
Comment (mathisscott) [inline src/clr-core/tag/tag.element.ts]: `danger` not `dander`

File: src/clr-core/badge/badge.element.ts (hunk 1 above)
Comment (coryrylan) [reply src/clr-core/badge/badge.element.ts]: Good point, I can add this to my todo list when I update the button from link to flat. I'll open a separate PR to fix both.

File: src/clr-core/tag/tag.element.ts (hunk 2 above)
Comment (coryrylan) [reply src/clr-core/tag/tag.element.ts]: fixed

**Comment by coryrylan** [comment]
Blocking on https://github.com/vmware/clarity/pull/4087 being merged first

**Pull request merged**